app/notifications/validators.py

```diff
@@ -57,31 +57,43 @@ def check_service_over_api_rate_limit(service, key_type):
                 raise RateLimitError(rate_limit, interval, key_type)
 
 
-def check_service_over_daily_message_limit(service, key_type):
+def check_service_over_daily_message_limit(service, key_type, notification_type=None):
     if key_type == KEY_TYPE_TEST or not current_app.config["REDIS_ENABLED"]:
         return 0
 
-    cache_key = daily_limit_cache_key(service.id)
+    limits = {
+        None: service.message_limit,
+        LETTER_TYPE: service.letter_message_limit,
+        EMAIL_TYPE: service.email_message_limit,
+        SMS_TYPE: service.sms_message_limit,
+    }
+    limit = limits[notification_type]
```

Comment:
for the notification types, this will be null for old cached SerialisedService objects and also for those services that haven't been migrated yet - this'll lead to us comparing an int to None and getting errors due to that

Reply:
Hng, serialised services, yep.